File: ziplink-front/src/services/user.service.js
import api from "./api"

const userService = {
    async getCurrentUser() {
        const response = await api.get("/users/me");
        return response.data.data;
    },

    async requestEmailChange(data) {
        const response = await api.post("/users/me/change-email", data);
        return response.data;
    },

    async deleteAccount(data) {
        const response = await api.delete("/users/me", { data });
        return response.data.data;
    },
};

export default userService;

File: ziplink/src/main/java/com/aryan/ziplink/service/UserService.java
package com.aryan.ziplink.service;

import com.aryan.ziplink.dto.request.ChangeEmailRequest;
import com.aryan.ziplink.dto.request.DeleteAccountRequest;
import com.aryan.ziplink.dto.response.UserResponse;

public interface UserService {
    UserResponse getCurrentUser();
    void requestEmailChange(ChangeEmailRequest request);
    void deleteAccount(DeleteAccountRequest request);
}
